src/main/java/com/chikli/hudson/plugin/naginator/NaginatorListener.java

```diff
@@ -28,91 +20,52 @@
         if (build instanceof MatrixRun) {
             return;
         }
-
-        // If we're supposed to check for a regular expression in the build output before
-        // scheduling a new build, do so.
-        if (naginator.isCheckRegexp()) {
-            LOGGER.log(Level.FINEST, "Got checkRegexp == true");
-
-            String regexpForRerun = naginator.getRegexpForRerun();
-            if ((regexpForRerun !=null) && (!regexpForRerun.equals(""))) {
-                LOGGER.log(Level.FINEST, "regexpForRerun - {0}", regexpForRerun);
-
-                try {
-                    // If parseLog returns false, we didn't find the regular expression,
-                    // so return true.
-                    if (!parseLog(build.getLogFile(), regexpForRerun)) {
-                        LOGGER.log(Level.FINEST, "regexp not in logfile");
-                        return;
-                    }
-                } catch (IOException e) {
-                    e.printStackTrace(listener
-                                      .error("error while parsing logs for naginator - forcing rebuild."));
-                }
-            }
-        }
-
-        if (canSchedule(build, naginator)) {
-            int n = naginator.getDelay().computeScheduleDelay(build);
-            LOGGER.log(Level.FINE, "about to try to schedule a build #{0} in {1} seconds for {2}",
-                    new Object[]{build.getNumber(), n, build.getProject().getName()} );
-            
-            List<Combination> combsToRerun = new ArrayList<Combination>();
-
-            if (naginator.isRerunMatrixPart()) {
-                if (build instanceof MatrixBuild) {
-                    MatrixBuild mb = (MatrixBuild) build;
-                    List<MatrixRun> matrixRuns = mb.getRuns();
-
-                    for(MatrixRun r : matrixRuns) {
-                        if (r.getNumber() == build.getNumber()) {
-                            if ((r.getResult() == SUCCESS) || (r.getResult() == ABORTED)) {
-                                continue;
+        
+        List<NaginatorScheduleAction> actions = build.getActions(NaginatorScheduleAction.class);
+        for (NaginatorScheduleAction action: actions) {
+            if (action.shouldSchedule(build, listener, retryCount)) {
+                int n = action.getDelay().computeScheduleDelay(build);
+                LOGGER.log(Level.FINE, "about to try to schedule a build #{0} in {1} seconds for {2}",
+                        new Object[]{build.getNumber(), n, build.getProject().getName()} );
+                
+                List<Combination> combsToRerun = new ArrayList<Combination>();
+    
+                if (action.isRerunMatrixPart()) {
+                    if (build instanceof MatrixBuild) {
+                        MatrixBuild mb = (MatrixBuild) build;
+                        List<MatrixRun> matrixRuns = mb.getRuns();
+    
+                        for(MatrixRun r : matrixRuns) {
```

Comment:
Oops. I added a commit to fix code formatting in 4e29dbb7f800c9399d12ce1f2f8cfcd2b0eb448a.